src/main/scala/rocket/RocketCore.scala

```diff
@@ -350,7 +395,9 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
     (io.rocc.busy || ex_reg_valid && ex_ctrl.rocc ||
      mem_reg_valid && mem_ctrl.rocc || wb_reg_valid && wb_ctrl.rocc)
   val id_csr_rocc_write = tile.roccCSRs.flatten.map(_.id.U === id_inst(0)(31,20)).orR && id_csr_en && !id_csr_ren
+  val id_vec_busy = io.vector.map(v => v.backend_busy || v.trap_check_busy).getOrElse(false.B)
   val id_do_fence = WireDefault(id_rocc_busy && (id_ctrl.fence || id_csr_rocc_write) ||
+    id_vec_busy && id_ctrl.fence ||
```

Comment:
why adding `id_ctrl.fence` here?

Reply:
Need to stall until there are no longer any vector memory operations in flight, so all preceding vector loads and stores become visible in the memory order.

Reply:
Nice, but it might alt the original behavior? e.g. w/o vector, `id_do_fence` will couple to `id_ctrl.fence`

Reply:
Perhaps it is clearer as 
```
(id_vec_busy && id_ctrl.fence) ||
```

W/o vector, `id_vec_busy` should be statically false, so that expression will never be true.